test/Microsoft.AspNetCore.Razor.Evolution.Test/DefaultDirectiveIRPassTest.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Evolution.Intermediate;
+using Xunit;
+using static Microsoft.AspNetCore.Razor.Evolution.Intermediate.RazorIRAssert;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public class DefaultDirectiveIRPassTest
+    {
+        [Fact]
+        public void Execute_MutatesIRDocument()
+        {
+            // Arrange
+            var content =
+@"@inherits Hello<World[]>
+@functions {
+    var value = true;
+}";
+            var originalIRDocument = Lower(content);
+            var pass = new DefaultDirectiveIRPass();
+
+            // Act
+            var irDocument = pass.Execute(codeDocument: null, irDocument: originalIRDocument);
+
+            // Assert
+            Assert.Same(originalIRDocument, irDocument);
+        }
+
+        [Fact]
+        public void Execute_Inherits_SetsClassDeclarationBaseType()
+        {
+            // Arrange
+            var content = "@inherits Hello<World[]>";
+            var originalIRDocument = Lower(content);
+            var pass = new DefaultDirectiveIRPass();
+
+            // Act
+            var irDocument = pass.Execute(codeDocument: null, irDocument: originalIRDocument);
+
+            // Assert
+            var @namespace = SingleChild<NamespaceDeclarationIRNode>(irDocument);
+            var @class = SingleChild<ClassDeclarationIRNode>(@namespace);
+            Assert.Equal(@class.BaseType, "Hello<World[]>");
+        }
+
+        [Fact]
+        public void Execute_Functions_ExistsAtClassDeclarationAndMethodLevel()
+        {
+            // Arrange
+            var content = "@functions { var value = true; }";
+            var originalIRDocument = Lower(content);
+            var pass = new DefaultDirectiveIRPass();
+
+            // Act
+            var irDocument = pass.Execute(codeDocument: null, irDocument: originalIRDocument);
+
+            // Assert
+            var @namespace = SingleChild<NamespaceDeclarationIRNode>(irDocument);
+            var @class = SingleChild<ClassDeclarationIRNode>(@namespace);
+            Children(@class,
+                node => Assert.IsType<RazorMethodDeclarationIRNode>(node),
+                node => CSharpStatement(" var value = true; ", node));
+            var method = (RazorMethodDeclarationIRNode)@class.Children[0];
+            Children(method,
+                node => Html(string.Empty, node),
+                node => Directive("functions", node,
+                    directiveChild => CSharpStatement(" var value = true; ", directiveChild)),
+                node => Html(string.Empty, node));
+        }
+
+        private static DocumentIRNode Lower(string content)
```

Comment:
IIRC I made a base class for this so it doesn't have to be copy pasted all over. If I didn't, please do 😆